File: server/use-cases/updateAction.ts
import prisma from "../utils/prisma"

import {
    UpdateActionParamsType,
} from '../../schemas/preventive'

export async function updateAction({ id, data }: UpdateActionParamsType) {
    try {
        await prisma.preventiveAction.update({
            where: { id },
            data
        })
    } catch (error) {
        throw error
    }
}

File: src-py/use_cases/update_action.py
from schemas.preventive import UpdateActionParams
from infra.prisma import prisma


async def updateAction(params: dict) -> None:
    validated = UpdateActionParams.model_validate(params)

    await prisma.preventiveaction.update(
        where={"id": validated.id},
        data=validated.data.model_dump(exclude_none=True),
    )
